resolwe/observers/models.py

```diff
@@ -57,32 +59,75 @@ def get_interested(
         query &= Q(object_id=object_id) | Q(object_id__isnull=True)
         return cls.objects.filter(query)
 
+    @classmethod
+    def observe_instance_container(
+        cls,
+        instance: PermissionObject,
+        change_type: ChangeType,
+        containers: Optional[Iterable[PermissionObject]] = None,
+    ):
+        """Handle a notifications to the containers of the given instance.
+
+        When containers are given notify them else infer them from the instance.
+        """
+        for container in containers or instance.containers:
+            observers = Observer.get_interested(
+                change_type=change_type,
+                content_type=ContentType.objects.get_for_model(container),
+                object_id=container.pk,
+            )
+            # Forward the message to the appropriate groups.
+            for subscriber in Subscription.objects.filter(observers__in=observers):
+                if container.has_permission(Permission.VIEW, subscriber.user):
+                    # Register on_commit callbacks to send the signals.
+                    Subscription.notify(
+                        subscriber.session_id,
+                        container,
+                        change_type,
+                        source=(
+                            ContentType.objects.get_for_model(instance).name,
+                            instance.id,
```

Comment:
Use `pk` instead of `id` to be consistent with the rest of the code

Reply:
Done.